### PR TITLE
Generate 2 projects when JWT Auth or rest Client is selected as Specification

### PR DESCRIPTION
**Pull request by rdebusscher**
to have more realistic examples

### CODE REVIEW

File: src/main/resources/files/Service.java.tpl [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package [# th:text="${java_package}"/].client;
+
+import org.eclipse.microprofile.rest.client.annotation.RegisterProvider;
```
Comment (Karm) [inline src/main/resources/files/Service.java.tpl]: @rdebusscher IMHO unused import

File: src/main/resources/files.lst [hunk 2]
```diff
@@ -1,14 +1,15 @@
...
 src/main/resources/files/kumuluzEE/RestApplication.java.tpl
+src/main/resources/files/kumuluzEE/config.yaml.tpl
+src/main/resources/files/kumuluzEE/service-b/RestApplication.java.tpl
+src/main/resources/files/kumuluzEE/service-b/config.yaml.tpl
+src/main/resources/files/MPJWTToken.java.tpl
+src/main/resources/files/Service.java.tpl
 src/main/resources/files/MetricController.java.tpl
-src/main/resources/files/ApplicationSecurityData.java.tpl
+src/main/resources/files/readme.md.top.tpl
 src/main/resources/files/thorntail-v2/web.xml.tpl
```
Comment (Karm) [inline src/main/resources/files.lst]: @rdebusscher  When I select MP 2.2 and Thorntail and *only* REST, no JWT, I still get this for service-a and I think it is superfluous:

```
karm@local:/tmp$ cat th-rest-demo/demo/service-a/src/main/webapp/WEB-INF/web.xml
<?xml version="1.0" encoding="UTF-8"?>
<web-app xmlns="http://xmlns.jcp.org/xml/ns/javaee"
         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://xmlns.jcp.org/xml/ns/javaee
		 http://xmlns.jcp.org/xml/ns/javaee/web-app_3_1.xsd" version="3.1">

    <security-constraint>
        <display-name>Secure All</display-name>
        <web-resource-collection>
            <web-resource-name>Secure REST Endpoint</web-resource-name>
            <url-pattern>/data/protected</url-pattern>
        </web-resource-collection>
        <auth-constraint>
            <role-name>*</role-name>
        </auth-constraint>
    </security-constraint>
```

Comment (rdebusscher) [reply src/main/resources/files.lst]: @Karm  Fixed in latest commit, thx!

File: src/main/resources/files/Service.java.tpl (hunk 1 above)
Comment (Karm) [reply src/main/resources/files/Service.java.tpl]: @rdebusscher  ?

Comment (rdebusscher) [reply src/main/resources/files/Service.java.tpl]: Missed this comment the first time. Done. Thx

File: src/main/java/org/eclipse/microprofile/starter/addon/microprofile/servers/server/TomeeServer.java [hunk 3]
```diff
@@ -82,5 +89,47 @@ public void adaptMavenModel(Model pomFile, JessieModel model) {
...
+        Xpp3Dom httpPort = new Xpp3Dom("tomeeHttpPort");
+        httpPort.setValue("8180");
+        configuration.addChild(httpPort);
+
+        Xpp3Dom shutdownPort = new Xpp3Dom("tomeeShutdownPort");
+        httpPort.setValue("8105");
+        configuration.addChild(shutdownPort);
+
+        Xpp3Dom ajpPort = new Xpp3Dom("tomeeAjpPort");
+        httpPort.setValue("8109");
```
[the remark this answers is not in the document]
Comment (rdebusscher) [reply src/main/java/org/eclipse/microprofile/starter/addon/microprofile/servers/server/TomeeServer.java]: Updated copy paste error.

**Comment by Karm** [comment]
@rdebusscher We need to rebase after 30 was merged.

 > This branch has conflicts that must be resolved

BTW, I have been playing with this branch and I was basing my Quarkus work on it here http://starter.karms.biz/ and I think the separation to service-a and service-b is good to go.
There are some superfluous files IMHO, but that could be resolved as improvement later.

**Comment by rdebusscher** [comment]
Merging this PR, we can improve on small details later on.

**Pull request merged**